### PR TITLE
[PVR] CPVRTimerInfoTag: Use std::optional for m_epgTag.

### PR DESCRIPTION
**Pull request by ksooo**
Just a small code modernization.

@phunkyfish please can you have a look.

### CODE REVIEW

File: xbmc/pvr/timers/PVRTimerInfoTag.h [hunk 1]
```diff
@@ -636,9 +637,8 @@ class CPVRTimerInfoTag final : public ISerializable
   unsigned int m_iRadioChildTimersRecording = 0;
   unsigned int m_iRadioChildTimersErrors = 0;
 
-  mutable std::shared_ptr<CPVREpgInfoTag> m_epgTag; /*!< epg info tag matching m_iEpgUid. */
+  mutable std::optional<std::shared_ptr<CPVREpgInfoTag>>
```
Comment (lrusak) [inline xbmc/pvr/timers/PVRTimerInfoTag.h:640]: I don't think this is really the correct use of std::optional. Wrapping a smart pointer can already be checked if it contains an object if it nullptr or not. Maybe I'm mistaken but I would think you use std::optional with an object or a smart pointer for delayed initialization, not both.

Comment (ksooo) [reply xbmc/pvr/timers/PVRTimerInfoTag.h:640]: > Wrapping a smart pointer can already be checked if it contains an object if it nullptr or not.

Sure, but not sufficient for my use case. I don't only want to implement lazy initialization, I want lazy eval in combination check only once. Legit result of the eval can be nullptr and exactly for this case I need the std::optional, to know that I already did the check.

File: xbmc/pvr/timers/PVRTimerInfoTag.cpp [hunk 2]
```diff
@@ -1336,14 +1333,18 @@ std::shared_ptr<CPVREpgInfoTag> CPVRTimerInfoTag::GetEpgInfoTag(bool bCreate /*
...
-              m_iEpgUid = m_epgTag->UniqueBroadcastID();
+              m_iEpgUid = (*m_epgTag)->UniqueBroadcastID();
           }
         }
       }
     }
-    m_bProbedEpgTag = true;
+    if (!m_epgTag)
+    {
+      // not an error; the tag might just be to old and is no longer available in cache/at client
```
Comment (arnova) [inline xbmc/pvr/timers/PVRTimerInfoTag.cpp:1343]: to -> too :-)

**Comment by ksooo** [comment]
Okay, the PR as it is is still buggy and although it can be fixed to work properly with a `std::optional` the resulting code is somehow hard to read. So, let's stick with what we have. I will close this one.

**Pull request closed without merging**